src/core/server/saved_objects/types.ts

```diff
@@ -82,6 +82,10 @@ export interface SavedObjectsFindOptions {
   search?: string;
   /** The fields to perform the parsed query against. See Elasticsearch Simple Query String `fields` argument for more information */
   searchFields?: string[];
+  /**
+   * Use the sort values from the previous page to retrieve the next page of results.
+   */
+  searchAfter?: Array<string | number>;
```

Comment:
Should we help the consumer with that new param? Looking at `src/core/server/saved_objects/service/lib/search_dsl/sorting_params.ts`, I'm not sure exactly what we would be supposed to use here. Maybe `#find()` should return a new field with the values from the last returned result to avoid having the consumer manually implement this logic?

Reply:
> Looking at src/core/server/saved_objects/service/lib/search_dsl/sorting_params.ts, I'm not sure exactly what we would be supposed to use here.

I typed this incorrectly and had to look into this further. The type should really be `unknown[]`, as the `sort` values in the response can vary based on the `sort` values provided in the request and whatever those fields are mapped to in ES. Usage looks like this:
```
// request
sort: [{ timestamp: 'asc', tiebreaker: 'asc' }]

// response
hits: [
  { _id: 'a', _source: { timestamp: 1612300559379 }, sort: [1612300559379, 'tiebreaker_a'] },
  { _id: 'b', _source: { timestamp: 1612300578340 }, sort: [1612300578340, 'tiebreaker_b'] },
]
```

> Maybe #find() should return a new field with the values from the last returned result to avoid having the consumer manually implement this logic?

I'm adding this now - my plan was to add the returned `sort` into the individual hits returned from `find` (similar to how we do for `score`). Then if someone wanted to get the next page, it would look something like this:

```ts
const { id } = await so.openPointInTimeForType('foo');

const page1 = await so.find({
  type: 'foo', // this gets ignored
  search: '...',
  pit: { id },
  sortField: '@timestamp',
  sortOrder: 'desc',
});
const lastHit = page1.saved_objects[page1.saved_objects.length - 1];

const page2 = await so.find({
  type: 'foo', // this gets ignored
  search: '...',
  pit: { id },
  sortField: '@timestamp',
  sortOrder: 'desc',
  searchAfter: lastHit.sort,
});
```

Since we are returning it with each hit, it more closely mirrors how ES does this. (Of course we could choose to add a new top-level field to `find` like `lastSort`, but this provides more flexibility as you can pull it from any field).

Reply:
>  my plan was to add the returned sort into the individual hits returned from find (similar to how we do for score)

This approach sounds good to me. I like the `searchAfter: lastHit.sort`, very elegant. 

(have fun fixing the TS failures associated with changing the `SavedObjectsFindResult` type though. Went there, done that 😅 )